cartographer_rviz/cartographer_rviz/drawable_submap.h

```diff
@@ -105,6 +109,7 @@ class DrawableSubmap : public QObject {
       GUARDED_BY(mutex_);
   float current_alpha_ = 0.f;
   std::unique_ptr<::rviz::BoolProperty> visibility_;
+  std::vector<bool> slices_visibility_;
```

Comment:
It is an invariant that slices_visibility_.size() == ogre_slices_.size(), right? Encode this, by pulling out a private struct `Slice` or move it into OgreSlice.